scripts/west_commands/run_common.py

```diff
@@ -186,7 +186,7 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
     if not user_args.skip_rebuild:
         rebuild(command, build_dir, user_args)
 
-    if domains is None:
+    if domain_file is None:
```

Comment:
My suggestion would be to drop `domains` as a function argument and just do:
```py
if user_args.domain is None and command.name != "flash":
    domains = [load_domains(build_dir).get_default_domain()]
else:
    domains = load_domains(build_dir).get_domains(user_args.domain,
                                                  default_flash_order=True)

```
Note that `default_flash_order=True` only has an effect when `user_args.domain is None`.

Reply:
Updated